bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/handler/ForecastSolarBridgeHandler.java

```diff
@@ -130,6 +134,13 @@ private synchronized void getData() {
         if (planes.isEmpty()) {
             return;
         }
+        if (calmDownEnd.isAfter(Instant.now(Utils.getClock()))) {
+            // wait until calm down time is expired
+            long minutes = java.time.Duration.between(Instant.now(Utils.getClock()), calmDownEnd).toMinutes();
```

Comment:
Maybe you can import java.time.Duration ?

That is the first time I see such pattern does not use the system clock. I am not sure what is the benefit. Would it be a problem to use Instant.now() here and this code in calmDown ?
```
         calmDownEnd = Instant.now().plus(CALM_DOWN_TIME_MINUTES, ChronoUnit.MINUTES);
```

Reply:
java.time.Clock is used only in 14 classes in all our bindings database and only in few bindings. energidataservice is the binding that uses it the most. @jlaur: can you please comment here, maybe I am missing something.

Reply:
The use-case of `Clock` is to be able to provide a pluggable representation of the current instant, which is quite convenient for unit tests and/or when you need to use a custom time-zone. As an example, the second constructor here is used by the unit tests:

https://github.com/openhab/openhab-addons/blob/86a8a07c143af7ce1d5307f41c3fd3e17da0f19b/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/CacheManager.java#L58-L63

It looks like the same is needed here. The `static` approach can be discussed, but I realize now that I may also have hardwired the system default time-zone longer than desired, because `CacheManager`'s life-cycle follows the handler's life-cycle.

Reply:
> It looks like the same is needed here.

Right, this was my intention. I can simulate the waiting time in unit testing.

> but I realize now that I may also have hardwired the system default time-zone

I pushed now a change where clock respects `TimeZoneProvider` all the time but for unit tests it's still possible to set a fixed `Clock` for test purposes.